### PR TITLE
feature: adding helper method FindContainedResource on DomainResource

### PR DESCRIPTION
**Pull request by marcovisserFurore**
Solves issue #180

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -161,7 +161,8 @@ src/Hl7.Fhir.Model/EwoutFhirNetApi.snk
 *.snk
```
Comment (ewoutkramer) [inline .gitignore]: Is this coming in from another fix?

File: src/Hl7.Fhir.Core/Model/DomainResource.cs [hunk 2]
```diff
@@ -56,5 +56,32 @@ public override IEnumerable<ValidationResult> Validate(ValidationContext validat
...
+        /// <param name="reference">the reference to contained resource</param>
+        /// <returns>The resource referenced by <paramref name="reference"/>, null otherwise.</returns>
+        public Resource FindContainedResource(ResourceReference reference)
+        {
+            if (!reference.IsContainedReference) return null;
+
+            if (reference.Reference == "#") return this;
+
+            // search the contained resource by removing '#' 
+            return Contained.FirstOrDefault(c => c.Id == reference.Reference.Remove(0, 1));
```
Comment (ewoutkramer) [inline src/Hl7.Fhir.Core/Model/DomainResource.cs]: This fails if reference.Reference is null.

File: .gitignore (hunk 1 above)
Comment (marcovisserFurore) [reply .gitignore]: The newest version of Visual Studio places a map called ``.vs/`` in the root folder. This has to be ignored for our project

File: src/Hl7.Fhir.Core/Model/DomainResource.cs (hunk 2 above)
Comment (marcovisserFurore) [reply src/Hl7.Fhir.Core/Model/DomainResource.cs]: Oops, you are right of course. I created a new commit for that

**Comment by ewoutkramer** [comment]
Fixes #180.

**Pull request merged**